src/lib/components/Access/utils.js

```diff
@@ -78,6 +127,57 @@ export function MessageSection({intent, icon, title, text}) {
 }
 
 
+class EmbargoCheckboxComponent extends Component {
+
+  render() {
+    const {
+      formik,
+      embargo
+    } = this.props;
+
+    return <Checkbox
+      disabled={embargo.is(EmbargoState.DISABLED)}
+      checked={embargo.is(EmbargoState.APPLIED)}
+      onChange={
+        (event, data) => {
+          if (formik.field.value) {
+            // NOTE: We reset values, so if embargo filled and user unchecks,
+            //       user needs to fill embargo again. Otherwise lots of
+            //       bookkeeping.
+            formik.form.setFieldValue("access.embargo", {
```

Comment:
Why not follow the same pattern as below? i.e `formik.form.setFieldValue("access.embargo.active", false);`

Reply:
... because of the comment above :smile_cat: ... but maybe the comment is not clear enough:

In this situation we need to wipe away the `until` and `reason` keys since the `Apply embargo` is not selected anymore. We do that by setting the whole object as opposed to a single field in the object as is done below. Otherwise we pass inconsistent embargo values under the hood and the backend will return an error that doesn't match up to what the user was expecting.  

In the case below, selecting the "Apply embargo" has no other responsibilities (it doesn't need to reset anything), so it just sets it directly.

Reply:
Ok I see!